lib/rubocop/cop/rspec/factory_bot/association_style.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Use consistent style to define associations.
+        #
+        # @safety
+        #   This cop may cause false-positives in `EnforcedStyle: explicit`
+        #   case. It recognizes any method call that has no arguments as an
+        #   implicit association but it might be a user-defined trait call.
+        #
+        # @example EnforcedStyle: implicit (default)
+        #   # bad
+        #   association :user
+        #
+        #   # good
+        #   user
+        #
+        # @example EnforcedStyle: explicit
+        #   # bad
+        #   user
+        #
+        #   # good
+        #   association :user
+        #
+        #   # good (defined in NonImplicitAssociationMethodNames)
+        #   skip_create
+        class AssociationStyle < Base
+          extend AutoCorrector
+
+          include ConfigurableEnforcedStyle
+
+          MSG = 'Use consistent style in FactoryBot associations.'
+
+          RESTRICT_ON_SEND = %i[
+            factory
+            trait
+          ].freeze
+
+          # @param node [RuboCop::AST::SendNode]
+          # @return [void]
+          def on_send(node)
+            bad_associations_in(node).each do |association|
+              add_offense(
+                association,
+                message: "Use #{style} style to define associations."
+              ) do |corrector|
+                autocorrect(corrector, association)
+              end
+            end
+          end
+
+          private
+
```

Comment:
Thanks, but I'll simply delete the unused `MSG` constant and leave the rest as is. This is because RuboCop insists that [`#message` should not be overridden](https://github.com/rubocop/rubocop/blob/722d1c1d70222306c8d63627ec81777025f5b29d/lib/rubocop/cop/base.rb#L100).